discovery-provider/src/challenges/listen_streak_challenge.py

```diff
@@ -0,0 +1,72 @@
+from collections import Counter
+from datetime import datetime
+from src.challenges.challenge import ChallengeManager, ChallengeUpdater
+from src.models import ListenStreakChallenge, Play
+from src.challenges.challenge_event import ChallengeEvent
+
+class ListenStreakChallengeUpdater(ChallengeUpdater):
+    """7-day listening streak challenge"""
+
+    def update_user_challenges(
+        self, session, event, user_challenges_metadata, step_count, extras
+    ):
+        # Destruct the list of user challenges from the list of tuples of (user_challenge, event_metadata)
+        user_challenges, _ = [list(t) for t in zip(*user_challenges_metadata)]
```

Comment:
Sorry this function signature was changed in https://github.com/AudiusProject/audius-protocol/pull/1657